### DISCUSSION TITLE
error adding symbols occurs when I cross-compile the strongswan

### DISCUSSION THREAD

**Discussion by AmieeChen** [strongswan/strongswan#307] [Q&A]
I tried to use MingGW on Ubuntu to cross-compile strongswan-5.2.0 for windows. After configuring, when I executed the command make, the error occured. The error information is as follows:

../../src/libcharon/plugins/vici/.libs/libvici.a: error adding symbols: Archive has no index; run ranlib to add one
collect2: error: ld returned 1 exit status
Makefile:588: recipe for target 'swanctl.exe' failed
make[4]: *** [swanctl.exe] Error 1
make[4]: Leaving directory '/home/xdwy/strongswan-5.2.0/src/swanctl'
Makefile:463: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/home/xdwy/strongswan-5.2.0/src/swanctl'
Makefile:490: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/home/xdwy/strongswan-5.2.0/src'
Makefile:552: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/xdwy/strongswan-5.2.0'
Makefile:460: recipe for target 'all' failed
make: *** [all] Error 2

when I executed the configure command, my command is:
CC=x86_64-w64-mingw32-gcc ./configure --host=x86_64-win64-mingw32  --disable-defaults --enable-monolithic --enable-static --enable-svc --enable-ikev2 --enable-ikev1 --enable-nonce --enable-pem --enable-pkcs1 --enable-x509 --disable-openssl --disable-socket-win --disable-kernel-wfp --disable-kernel-iph --enable-pubkey --enable-swanctl --with-swantldir=swanctl --with-strongswan-conf=strongswan.conf 

Besides, I executed the command  ranlib libvici.a and reconfigured. However, the error was the same. I want to know how to solve this problem. I appreciate that any one can give me some suggestions.

**Comment by Thermi** [comment]
Try cleaning the source directory (`make clean`) and then building again. Please make sure this is not a false positive. Also 5.2.0 is very, very old.

**Reply by AmieeChen** [reply]
Thans for your suggestion. I also tried to cross-compile the latest version: strongswan-5.9.2, but the same error occured.
